src/HasAttributes.php

```
@@ -130,4 +132,30 @@ public function removeAttribute(string|UnitEnum $key): static
+     * @param string $prefix Optional prefix to prepend to the key (for example, `aria-`, `data-`, `on-`).
+     * @param bool $boolToString Whether to convert boolean values to `true` and `false` strings.
+     *
+     * @return static New instance with the updated `attributes` value.
+     *
+     * @phpstan-param scalar|Stringable|UnitEnum|Closure(): mixed $value
+     */
+    public function setAttribute(
+        string|UnitEnum $key,
+        bool|float|int|string|Closure|Stringable|UnitEnum|null $value,
+        string $prefix = '',
+        bool $boolToString = false,
```

PHPMD (2.15.0)

[error] The method setAttribute has a boolean flag argument $boolToString, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)